llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -4205,6 +4218,76 @@ void CombinerHelper::applyExtractVecEltBuildVec(MachineInstr &MI,
   replaceSingleDefInstWithReg(MI, Reg);
 }
 
+bool CombinerHelper::matchCombineExtractToShuffle(
```

Comment:
I suggest to follow along with the DAG version; you'll want something similar to buildLegalVectorShuffle, which will avoid creating shuffles which will just be broken up again https://github.com/llvm/llvm-project/blob/33c14f19656c751bbbc083e4a168ab898e583bfd/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp#L23501

Reply:
Thanks for the suggestion, I am now avoiding making a shufflevector after input from Amara. But I will be sure to check this if I need to make a shufflevector in the future.